main.nf

```diff
@@ -272,13 +272,14 @@ aligned.filter { logs, bams -> check_log(logs) }
     SPLIT_BAMS.into {bam_count; bam_rseqc; bam_preseq; bam_markduplicates; bam_featurecounts; bam_stringtieFPKM}
 
 //Counts the number of bam files from STAR. Uses this to determine whether to run 'sampleCorrelation' process or not
-bam_count.count()
-    .subscribe { if (count.toFloat() <= '3'.toFloat()){
-                    params.sampleLevel = true
-                    } else {
-                    params.sampleLevel = false
-                    }
-
+if (isNull(params.sampleLevel)){
+    bam_count.count()
```

Comment:
Does it matter that `bam_count` is now not consumed if `sampleLevel` is specified? Should be fine, right? Won't make nextflow hang?